pkgs/os-specific/linux/shadow/default.nix

```diff
@@ -6,7 +6,7 @@
 , libxcrypt
 , glibcCross ? null
 , pam ? null
-, withTcb ? stdenv.isLinux, tcb
+, withTcb ? lib.meta.availableOn stdenv.hostPlatform tcb, tcb
```

Comment:
The problem with this is, that when we change the platform of tcb for some reason, we could trigger a mass rebuild or silently disable a feature without realizing it. Using isGnu here would duplicate code but IMO I would prefer it.